### PR TITLE
Azure lib changes in data buckets flow - Part 3/4

### PR DESCRIPTION
**Pull request by romayalon**
Signed-off-by: Romy <romy2232@gmail.com>

### Explain the changes
1. Replace old azure lib function calls with the new lib function calls in agent.js, block_store_azure and block_store_client.
2. Remove send_logs file.
3. lambda funcs - sync_s3_to_azure - replace old azure lib function calls with the new lib function calls.

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1.

### CODE REVIEW

File: src/agent/block_store_services/block_store_azure.js [hunk 1]
```diff
@@ -151,31 +137,33 @@ class BlockStoreAzure extends BlockStoreBase {
...
-                    this.base_path,
-                    prev_continuation_token,
-                    callback));
-                if (list_res.entries.length !== 0) {
-                    await P.map_with_concurrency(10, list_res.entries, async entry => {
+                let iterator = this.container_client.listBlobsFlat({
+                    prefix: this.base_path
+                }).byPage({ continuationToken: prev_continuation_token, maxPageSize: 1000 });
+                let list_res = (await iterator.next()).value;
+                dbg.log1('block_store_azure.cleanup_target_path list_res: ', list_res);
```
Comment (liranmauda) [inline src/agent/block_store_services/block_store_azure.js:144]:
```suggestion
                const iterator = this.container_client.listBlobsFlat({
                    prefix: this.base_path
                }).byPage({ continuationToken: prev_continuation_token, maxPageSize: 1000 });
                const list_res = (await iterator.next()).value;
                dbg.log1('block_store_azure.cleanup_target_path list_res: ', list_res);
```

File: src/agent/block_store_services/block_store_azure.js [hunk 2]
```diff
@@ -192,38 +180,36 @@ class BlockStoreAzure extends BlockStoreBase {
             count: 0
         };
         let failed_to_delete_block_ids = [];
-        return P.map_with_concurrency(10, block_ids, block_id => {
+        dbg.log0('block_store_azure._delete_blocks block_ids: ', block_ids);
```
Comment (liranmauda) [inline src/agent/block_store_services/block_store_azure.js:183]:
```suggestion
        dbg.log1('block_store_azure._delete_blocks block_ids: ', block_ids);
```

File: src/agent/block_store_services/block_store_azure.js [hunk 3]
```diff
@@ -192,38 +180,36 @@ class BlockStoreAzure extends BlockStoreBase {
...
+
+        return P.map_with_concurrency(10, block_ids, async block_id => {
                 const block_key = this._block_key(block_id);
                 let info;
```
Comment (liranmauda) [inline src/agent/block_store_services/block_store_azure.js:187]: unnecessary
```suggestion
                let info;
```

File: src/agent/block_store_services/block_store_azure.js [hunk 4]
```diff
@@ -192,38 +180,36 @@ class BlockStoreAzure extends BlockStoreBase {
...
-                        }
-                        dbg.warn('BlockStoreAzure _delete_blocks failed for block',
-                            this.container_name, block_key, err);
-                    });
+                try {
+                    const blob_client = azure_storage.get_blob_client(this.container_client, block_key);
+                    const info_arg = await blob_client.getProperties();
+                    dbg.log1('block_store_azure._delete_blocks info_arg: ', info_arg, 'block_key: ', block_key);
+
+                    info = info_arg;
```
Comment (liranmauda) [inline src/agent/block_store_services/block_store_azure.js:193]:
```suggestion
                   const  info = info_arg;
```

File: src/agent/block_store_services/block_store_azure.js [hunk 5]
```diff
@@ -287,25 +271,23 @@ class BlockStoreAzure extends BlockStoreBase {
     }
 
     _read_usage() {
```
Comment (liranmauda) [inline src/agent/block_store_services/block_store_azure.js:273]: maybe make it async?

File: src/agent/block_store_services/block_store_client.js [hunk 6]
```diff
@@ -241,26 +242,29 @@ class BlockStoreClient {
...
                 const block_key = `${bs_info.blocks_path}/${block_dir}/${block_id}`;
-                const info = await P.fromCallback(callback => blob.getBlobToStream(container, block_key, writable, {
-                        disableContentMD5Validation: true
-                    },
-                    callback));
+                const container_client = azure_storage.get_container_client(blob, container);
+                const blob_client = azure_storage.get_blob_client(container_client, block_key);
+                dbg.log1('block_store_client._delegate_read_block_azure starting download block_key:', block_key, ' container: ', container);
+
+                const info = await blob_client.download(0, undefined);
```
Comment (liranmauda) [inline src/agent/block_store_services/block_store_client.js:252]: maybe call it response instead of info?

**Review by liranmauda — CHANGES_REQUESTED** [review]

File: src/agent/block_store_services/block_store_azure.js (hunk 4 above)
Comment (romayalon) [reply src/agent/block_store_services/block_store_azure.js]: I just delete the info and used info_arg instead

**Review by liranmauda — APPROVED** [review]
LGTM

**Pull request merged**